### PR TITLE
[AIR] Fix examples and tests to specify `batch_format` in `BatchMapper`

### LINKED ISSUE #30431: [AIR] [CI] BatchMapper needs to have `batch_format` set in all CI tests/examples
### What happened + What you expected to happen

https://github.com/ray-project/ray/pull/30366 removed the default pandas batch format, and users are now required to specify the format in `BatchMapper(..., batch_format=...)`.

### Versions / Dependencies

master

### Reproduction script

These are the tests that need to have `batch_format` specified:

Screen Shot 2022-11-17 at 2 16 54 PM

### Issue Severity

Medium: It is a significant difficulty but I can work around it.

### PR DESCRIPTION
**Pull request by justinvyu**
## Why are these changes needed?

Followup to https://github.com/ray-project/ray/pull/30366 to update remaining tests, examples, and doc code to specify a batch format (setting it to `pandas`, which was the default before).

## Related issue number

Closes https://github.com/ray-project/ray/issues/30431

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by amogkam — APPROVED** [review]
Thanks for doing this @justinvyu!

File: python/ray/train/batch_predictor.py [hunk 1]
```diff
@@ -287,7 +287,9 @@ def __call__(self, input_batch: DataBatchType) -> DataBatchType:
             if preprocessor:
                 # Set the in-predictor preprocessing to a no-op when using a separate
                 # GPU stage. Otherwise, the preprocessing will be applied twice.
-                override_prep = BatchMapper(lambda x: x)
+                override_prep = BatchMapper(
+                    lambda x: x, batch_format=predict_stage_batch_format
+                )
```
Comment (justinvyu) [inline python/ray/train/batch_predictor.py:292]: Is it possible to add a unit test for this codepath? Test preprocessing before sending to a separate GPU stage to do prediction.

Comment (amogkam) [reply python/ray/train/batch_predictor.py:292]: this is being tested here https://sourcegraph.com/github.com/ray-project/ray/-/blob/python/ray/train/tests/test_batch_predictor.py?L104:5

Comment (justinvyu) [reply python/ray/train/batch_predictor.py:292]: I see, I think the test was still pending and not showing up on the flaky tests earlier.

**Review by richardliaw — APPROVED** [review]

**Pull request merged**